agent/testing/testing.go

```diff
@@ -0,0 +1,85 @@
+/*
+   ToDD testing package
+
+   Contains infrastructure running testlets as well as maintaining
+   conformance for other native-Go testlet projects
+
+   Copyright 2016 Matt Oswalt. Use or modification of this
+   source code is governed by the license provided here:
+   https://github.com/Mierdin/todd/blob/master/LICENSE
+*/
+
+package testing
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+var (
+
+	// This map provides name redirection so that the native testlets can use names that don't
+	// conflict with existing system tools (i.e. using "toddping" instead of "ping") but users
+	// can still refer to the testlets using simple names.
+	//
+	// In short, users refer to the testlet by <key> and this map will redirect to the
+	// actual binary name <value>
+	nativeTestlets = map[string]string{
+		"ping": "toddping",
+	}
+)
+
+// Testlet defines what a testlet should look like if built in native
+// go and compiled with the agent
+type Testlet interface {
+
+	// Run is the "workflow" function for a testlet. All testing takes place here
+	// (or in a function called within)
+	//
+	// Params are
+	// target (string)
+	// args ([]string)
+	// timeLimit (int in seconds)
+	//
+	// Returns:
+	// metrics (map[string]string)
+	// (name of metric is key, value is metric value)
+	Run(string, []string, int) (map[string]string, error)
```

Comment:
You're totally right - this is a result of me still not sure about what to do with docstrings (I am still very used to sphinx-compatible docstring standards in Python). Will fix.